packages/vite/src/node/server/index.ts

```diff
@@ -129,28 +130,37 @@ export interface ServerOptions {
 }
 
 export interface ResolvedServerOptions extends ServerOptions {
-  fsServe: Required<FileSystemServeOptions>
+  fsServe: Required<Omit<FileSystemServeOptions, 'root'>>
```

Comment:
Is this wanted? I think this will throw compile time errors in users configuration instead of telling them that it is deprecated 🤔

Reply:
This is "resolved". The user side is marked as deprecated using jsdocs.